cmd/tendermint/commands/reindex_event.go

```diff
@@ -27,58 +28,70 @@ const (
 )
 
 // ReIndexEventCmd allows re-index the event by given block height interval
-var ReIndexEventCmd = &cobra.Command{
-	Use:   "reindex-event",
-	Short: "reindex events to the event store backends",
-	Long: `
+func MakeReindexEventCommand(conf *tmcfg.Config, logger log.Logger) *cobra.Command {
+	var (
+		startHeight int64
+		endHeight   int64
+	)
+
+	cmd := &cobra.Command{
+		Use:   "reindex-event",
+		Short: "reindex events to the event store backends",
+		Long: `
 reindex-event is an offline tooling to re-index block and tx events to the eventsinks,
 you can run this command when the event store backend dropped/disconnected or you want to 
 replace the backend. The default start-height is 0, meaning the tooling will start 
 reindex from the base block height(inclusive); and the default end-height is 0, meaning 
 the tooling will reindex until the latest block height(inclusive). User can omit
 either or both arguments.
 	`,
-	Example: `
+		Example: `
 	tendermint reindex-event
 	tendermint reindex-event --start-height 2
 	tendermint reindex-event --end-height 10
 	tendermint reindex-event --start-height 2 --end-height 10
 	`,
-	Run: func(cmd *cobra.Command, args []string) {
-		bs, ss, err := loadStateAndBlockStore(config)
-		if err != nil {
-			fmt.Println(reindexFailed, err)
-			return
-		}
-
-		if err := checkValidHeight(bs); err != nil {
-			fmt.Println(reindexFailed, err)
-			return
-		}
+		Run: func(cmd *cobra.Command, args []string) {
+			bs, ss, err := loadStateAndBlockStore(conf)
+			if err != nil {
+				fmt.Println(reindexFailed, err)
+				return
+			}
 
-		es, err := loadEventSinks(config)
-		if err != nil {
-			fmt.Println(reindexFailed, err)
-			return
-		}
+			cvhArgs := checkValidHeightArgs{
+				startHeight: startHeight,
+				endHeight:   endHeight,
+			}
+			if err := checkValidHeight(bs, cvhArgs); err != nil {
+				fmt.Println(reindexFailed, err)
+				return
+			}
 
-		if err = eventReIndex(cmd, es, bs, ss); err != nil {
-			fmt.Println(reindexFailed, err)
-			return
-		}
+			es, err := loadEventSinks(conf)
+			if err != nil {
+				fmt.Println(reindexFailed, err)
```

Comment:
Not you, but, any reason not to use `RunE` and return error here instead? `Println` seems a little odd. Might as well correct while we're here.